src/components/password/password.spec.tsx

```diff
@@ -0,0 +1,201 @@
+      expect(wrapper.find("input").at(0).prop("type")).toBe("password");
+    });
+
+    it("input type should change from `password` to `text` on click", () => {
+      act(() => {
+        wrapper.find(StyledButtonMinor).simulate("click", mockEvent);
+      });
+      wrapper.update();
+      expect(wrapper.find("input").at(0).prop("type")).toBe("text");
+    });
+
+    it("autoComplete attribute is `off`", () => {
+      expect(wrapper.find("input").at(0).prop("autoComplete")).toBe("off");
+    });
+
+    it.each([
+      ["foo", "foo"],
+      [mockedGuid, ""],
+    ])(
+      "Password input id should be custom value or guid depending on passed value",
+      (idType, id) => {
+        wrapper = renderPasswordInput({ id });
+        expect(wrapper.find("input").at(0).prop("id")).toBe(idType);
+      }
+    );
+  });
+
+  describe("buttonMinor prop checks", () => {
+    it("aria-controls attribute is correct", () => {
+      wrapper = renderPasswordInput({ id: "foo" });
+      expect(wrapper.find(StyledButtonMinor).prop("aria-controls")).toBe("foo");
+    });
+
+    it("buttonType attribute is correct", () => {
+      expect(wrapper.find(StyledButtonMinor).prop("buttonType")).toBe(
+        "tertiary"
+      );
+    });
+
+    it("iconPosition attribute is correct", () => {
+      expect(wrapper.find(StyledButtonMinor).prop("iconPosition")).toBe(
+        "before"
+      );
+    });
+
+    it("size attribute is correct", () => {
+      expect(wrapper.find(StyledButtonMinor).prop("size")).toBe("small");
+    });
+
+    it("default label text should be 'Show`", () => {
+      expect(wrapper.find(StyledButtonMinor).text()).toBe("Show");
+    });
+
+    it("label text should change from `Show` to `Hide` onClick", () => {
+      act(() => {
+        wrapper.find(StyledButtonMinor).simulate("click", mockEvent);
+      });
+      expect(wrapper.find(StyledButtonMinor).text()).toBe("Hide");
+    });
+
+    it("default aria-label text should be 'Show password`", () => {
+      expect(wrapper.find(StyledButtonMinor).prop("aria-label")).toBe(
+        "Show password"
+      );
+    });
+
+    it("aria-label text should change from `Show Password` to `Hide password` onClick", () => {
+      act(() => {
+        wrapper.find(StyledButtonMinor).simulate("click", mockEvent);
+      });
+      expect(wrapper.find(StyledButtonMinor).prop("aria-label")).toBe(
+        "Hide password"
+      );
+    });
+
+    it("default iconType should be 'view'", () => {
+      act(() => {
+        wrapper.find(StyledButtonMinor).simulate("click", mockEvent);
+      });
+      expect(wrapper.find(StyledButtonMinor).prop("iconType")).toBe("view");
+    });
+
+    it("iconType should change from `show` to `hide` onClick", () => {
+      act(() => {
+        wrapper.find(StyledButtonMinor).simulate("click", mockEvent);
+      });
+      expect(wrapper.find(StyledButtonMinor).prop("iconType")).toBe("hide");
+    });
+
+    it("default styling should be correct", () => {
```

Comment:
Might be a good idea to put these styling tests inside their own `describe` block.